### DISCUSSION TITLE
Can't change the SoftAP

### DISCUSSION THREAD

**Discussion by Tipmanfrank** [Hieromon/AutoConnect#367] [How can I?] [answered]
Hi

I am using below code for Autoconnect. What do I wrong since I can't change the SoftAP. All the time I see the esp32ap.
I expected that Config.apid would change the name of the ap.

#include <WiFi.h>          // Replace with WiFi.h for ESP32
#include <WebServer.h>     // Replace with WebServer.h for ESP32
#include <AutoConnect.h>

WebServer Server;          // Replace with WebServer for ESP32
AutoConnect      Portal(Server);
AutoConnectConfig Config;

void rootPage() {
  char content[] = "Hello, world";
  Server.send(200, "text/plain", content);
}

void setup() {
  delay(1000);
  Serial.begin(115200);
  Serial.println();
  Config.apid = "ap_portal";
  Config.psk = "12345678";
  Config.autoReconnect = true;

  Server.on("/", rootPage);
  if (Portal.begin()) {
    Serial.println("WiFi connected: " + WiFi.localIP().toString());
  }
}

void loop() {
    Portal.handleClient();
}

**Comment by Hieromon** [comment, marked as answer]
You are missing `Portal.config(Config);`. It's easy to forget, so I mentioned it in the documentation.
https://hieromon.github.io/AutoConnect/advancedusage.html

**Reply by Tipmanfrank** [reply]
Thanks this worked. I have been reading the documentation but I didn't see it.

By the way. After the ESP is connected to my local wifi I can't see its SoftAP anymore. That's ok. But how can I get in contact with the ESP webserver? I need a webserver to run locally on the ESP but also need the ESP to connect to Wifi and get data from internet. How can I connect to the ESP webserver, when I can't see the SoftAP? I don't know the IP address on the local network. 

Is there a way I always can connect to the SoftAP?

**Reply by Hieromon** [reply]
After the ESP module establishes a connection to the access point, you have three options to access the launched webserver on that ESP module.
However, if you're expecting the captive portal screen to pop up automatically even after the ESP module can connect to the some access point, you can't. (The captive portal screen pops up automatically because it has no transparency with the Internet. It is caused by the mechanism of the client OS. In other words, if the ESP module has a connection with the internet, the client OS will not trigger an automatic pop. AutoConnect is neither a router nor a gateway)

1. Use mDNS responder
    iOS, OSX, Linux and Win10 support [Multicast DNS](https://en.wikipedia.org/wiki/Multicast_DNS). By starting the mDNS service after successfully connecting to the access point with *AutoConnect::begin*, clients in the same segment can access the ESP module by name like `http://esp32-xxx.local`. This way you do not need to know in advance the dynamic IP address that the AP has assigned to the ESP module.
    [Refer to Combination with mDNS documentation]( https://hieromon.github.io/AutoConnect/adnetwork.html#combination-with-mdns).
2. Assign a static IP address to the ESP module in advance
    The most reliable method. See [Static IP assignment as a client](https://hieromon.github.io/AutoConnect/adnetwork.html#static-ip-assignment-as-a-client) section in the documentation for how to assign a static IP address to the ESP module.
 3. Use [preserveAPMode](https://hieromon.github.io/AutoConnect/adconnection.html#preserve-ap-mode) with  [retainPortal](https://hieromon.github.io/AutoConnect/apiconfig.html#retainportal)
    Use **retainPortal** to maintain SoftAP even after the ESP module is transparent to the Internet. But that's not enough. Even if you can connect in STA mode, if you always want to run AP mode just to know the IP address, you need to **disable** [**autoRise**](https://hieromon.github.io/AutoConnect/adcpcontrol.html#disable-the-captive-portal) and start AP by the sketch program.
Therefore, according to this method, the captive portal screen does not pop up automatically.

**Reply by Tipmanfrank** [reply]
Thanks a lot. I will go with solution 1. Now I will build the webserver. I need some input values from this stored into Eprom. After that I will contact a mysql server to data from there, too.
I am building a horn for sea races. Different times shall be possible to set locally in the ESP. But it shall also be possible to get information from my database driven by a homepage.
